src/main/java/io/cdap/plugin/switchcase/route/RoutingSwitch.java

```diff
@@ -0,0 +1,348 @@
+      String error = String.format("Record contained unknown value for field %s", config.routingField);
+      InvalidEntry<StructuredRecord> invalid = new InvalidEntry<>(1, error, input);
+      emitter.emitError(invalid);
+    } else if (Config.DefaultHandling.SKIP == config.getDefaultHandling()) {
+      LOG.trace("Skipping record because value for field {} did not match any rule in the port specification",
+                config.routingField);
+    }
+  }
+
+  private void emitNullValue(StructuredRecord input, MultiOutputEmitter<StructuredRecord> emitter) {
+    if (Config.NullHandling.NULL_PORT == config.getNullHandling()) {
+      emitter.emit(config.nullPort, input);
+    } else if (Config.NullHandling.ERROR_PORT == config.getNullHandling()) {
+      String error = String.format("Record contained null value for field %s", config.routingField);
+      InvalidEntry<StructuredRecord> invalid = new InvalidEntry<>(2, error, input);
+      emitter.emitError(invalid);
+    } else if (Config.NullHandling.SKIP == config.getNullHandling()) {
+      LOG.trace("Skipping record because field {} is null", config.routingField);
+    }
+  }
+
+  private Map<String, Schema> generateSchemas(Schema inputSchema) {
+    Map<String, Schema> schemas = new HashMap<>();
+    // Add all ports from the port config to the schemas
+    for (String port : evaluator.getAllPorts()) {
+      schemas.put(port, inputSchema);
+    }
+    // If defaulting records need to be sent to their own port, add that port to the schemas
+    if (Config.DefaultHandling.DEFAULT_PORT == config.getDefaultHandling()) {
+      schemas.put(config.defaultPort, inputSchema);
+    }
+    // If null values need to be sent to their own port, add that port to the schemas
+    if (Config.NullHandling.NULL_PORT == config.getNullHandling()) {
+      schemas.put(config.nullPort, inputSchema);
+    }
+    return schemas;
+  }
+
+  /**
+   * {@link PluginConfig} for {@link RoutingSwitch}
+   */
+  static class Config extends PluginConfig {
+    private static final List<Schema.Type> ALLOWED_TYPES = new ArrayList<>();
+    static {
+      ALLOWED_TYPES.add(Schema.Type.STRING);
+      ALLOWED_TYPES.add(Schema.Type.INT);
+      ALLOWED_TYPES.add(Schema.Type.LONG);
+      ALLOWED_TYPES.add(Schema.Type.FLOAT);
+      ALLOWED_TYPES.add(Schema.Type.DOUBLE);
+      ALLOWED_TYPES.add(Schema.Type.BOOLEAN);
+    }
+    @VisibleForTesting
+    static final String DEFAULT_PORT_NAME = "Default";
+    @VisibleForTesting
+    static final String DEFAULT_NULL_PORT_NAME = "Null";
+
+    private static final String ROUTING_FIELD_PROPERTY_NAME = "routingField";
+    static final String BASIC_PORT_SPECIFICATION_PROPERTY_NAME = "portSpecification";
+    private static final String DEFAULT_HANDLING_PROPERTY_NAME = "defaultHandling";
+    private static final String NULL_HANDLING_PROPERTY_NAME = "nullHandling";
+
+    //TODO: Add support for mode to switch between basic and jexl
```

Comment:
we should have a jira for it.

Reply:
Planning to add this right after this PR is merged. Have the code, just didn't want to include everything in one giant PR, so created these placeholders.